tests/Aql/HashedCollectExecutorTest.cpp

```diff
@@ -49,674 +53,573 @@ namespace arangodb {
-  HashedCollectExecutor testee(fetcher, infos);
-
-  OutputAqlItemRow result(std::move(block), infos.getOutputRegisters(),
-                          infos.registersToKeep(), infos.registersToClear());
-  std::tie(state, stats) = testee.produceRows(result);
-  ASSERT_EQ(state, ExecutionState::DONE);
-  ASSERT_FALSE(result.produced());
+template <size_t... vs>
+const HashedCollectSplitType splitIntoBlocks =
+    HashedCollectSplitType{std::vector<std::size_t>{vs...}};
+template <size_t step>
+const HashedCollectSplitType splitStep = HashedCollectSplitType{step};
+
+INSTANTIATE_TEST_CASE_P(HashedCollect, HashedCollectExecutorTest,
+                        ::testing::Combine(::testing::Values(splitIntoBlocks<2, 3>,
+                                                             splitIntoBlocks<3, 4>,
+                                                             splitStep<1>, splitStep<2>),
+                                           ::testing::Bool()));
+
+// Collect with only one group value
+TEST_P(HashedCollectExecutorTest, collect_only) {
+  auto infos = buildInfos(1, 2, {{1, 0}});
+  AqlCall call{};          // unlimited produce
+  ExecutionStats stats{};  // No stats here
+  ExecutorTestHelper<HashedCollectExecutor>(*fakedQuery)
+      .setInputValue({{{1}}, {{1}}, {{2}}, {{1}}, {{6}}, {{2}}, {{R"("1")"}}})
+      .setInputSplitType(getSplit())
+      .setCall(call)
+      .expectOutput({1}, {{1}, {2}, {6}, {R"("1")"}})
+      .allowAnyOutputOrder(true)
+      .expectSkipped(0)
+      .expectedState(ExecutionState::DONE)
+      .appendEmptyBlock(appendEmpty())
+      // .expectedStats(stats)
+      .run(std::move(infos));
 }
 
-TEST_F(HashedCollectExecutorTestNoRows, the_producer_waits) {
-  SingleRowFetcherHelper<::arangodb::aql::BlockPassthrough::Disable> fetcher(itemBlockManager, input.steal(), true);
-  HashedCollectExecutor testee(fetcher, infos);
-
-  OutputAqlItemRow result(std::move(block), infos.getOutputRegisters(),
-                          infos.registersToKeep(), infos.registersToClear());
-  std::tie(state, stats) = testee.produceRows(result);
-  ASSERT_EQ(state, ExecutionState::WAITING);
-  ASSERT_FALSE(result.produced());
-
-  std::tie(state, stats) = testee.produceRows(result);
-  ASSERT_EQ(state, ExecutionState::DONE);
-  ASSERT_FALSE(result.produced());
+// Collect skip all
+TEST_P(HashedCollectExecutorTest, skip_all) {
+  auto infos = buildInfos(1, 2, {{1, 0}});
+  AqlCall call{};
+  call.offset = 1000;      // skip all
+  ExecutionStats stats{};  // No stats here
+  ExecutorTestHelper<HashedCollectExecutor>(*fakedQuery)
+      .setInputValue({{{1}}, {{1}}, {{2}}, {{1}}, {{6}}, {{2}}, {{R"("1")"}}})
+      .setInputSplitType(getSplit())
+      .setCall(call)
+      .expectOutput({1}, {})
+      .allowAnyOutputOrder(true)
+      .expectSkipped(4)
+      .expectedState(ExecutionState::DONE)
+      .appendEmptyBlock(appendEmpty())
+      // .expectedStats(stats)
```

Comment:
can be removed?

Reply:
I would like to keep them around @goedderz has perpared some Stats Checking that can be simply activated.